### PR TITLE
Update counter component docs

### PR DESCRIPTION
**Pull request by mvn23**
**Description:**
Add some information about the initial/restore config options (ref. #6479, home-assistant/home-assistant#17101)
Convert counter configuration description to new format (ref. #6385)

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** N/A

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/counter.markdown [hunk 1]
```diff
@@ -24,18 +24,45 @@ counter:
...
+      default: True
+    step:
+      description: Incremental/step value for the counter.
+      required: false
+      type: integer
+      default: 1
+    icon:
+      description: Icon to display for the counter.
+      required: false
+      type: icon
```
Comment (WofWca) [inline source/_components/counter.markdown:56]: `icon` type?

Comment (mvn23) [reply source/_components/counter.markdown:56]: That's what I found in various other pages (input_boolean, input_number, input_select)

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks for this enhancement @mvn23 🏅

File: source/_components/counter.markdown (hunk 1 above)
Comment (fabaff) [reply source/_components/counter.markdown:56]: `icon:` is now handled like `device_class:` (#6495). Thanks for pointing it out @WofWca.

**Pull request merged**